server/keyshare/email.go

```diff
@@ -129,3 +135,12 @@ func sendHTMLEmail(addr string, a smtp.Auth, from, to *mail.Address, subject str
 		"\r\n")
 	return smtp.SendMail(addr, a, from.Address, []string{to.Address}, append(headers, msg...))
 }
+
+func verifyMXRecord(email string) error {
+	at := strings.LastIndex(email, "@")
+	_, err := net.LookupMX(email[at+1:])
+	if err != nil {
+		return err
+	}
```

Comment:
I'm not sure whether there can be cases that err == nil, but the list of MX records is empty. Did you check that?

Since the MX records list is always initialised, you could also do something like this:
```
records, _ := net.LookupMX(email[at+1:])
if len(records) == 0 {
	return ErrInvalidEmail
}
```
Then we circumvent that issue.

Another little thing, `strings.LastIndex` returns -1 if the string does not contain a `@`. -1+1=0, so in practice your function would crash. But in coding it's nice if you make these things explicit to make clear that you've taken this into account. For example, first do a `if at < 0 { ... }` check.

Reply:
> I'm not sure whether there can be cases that err == nil, but the list of MX records is empty. Did you check that?

verified, a domain witn no MX records causes an error

Reply:
> in coding it's nice if you make these things explicit

done